src/directedGraph.js

```diff
@@ -0,0 +1,84 @@
+export default class DirectedGraph {
+  constructor(){
+    this.index = 0
+    this.graph = {}
+  }
+
+  addVertex(name){
+    this.graph[name] = {'name':name,paths:{}}
+    this.index++
+  }
+
+  addDirection(v1, v2, weight){
+    this.graph[v1]['paths'][v2] = weight
+  }
+
+  hasVertex(name){
+    return this.graph[name] ? true : false
+  }
+
+  hasDirection(v1, v2){
+    return this.graph[v1]['paths'][v2] ? true : false
+  }
+
+  getDirectionWeight(v1, v2){
+    return this.graph[v1]['paths'][v2]
+  }
+
+  visit(v1, action){
+    let toCheck = [this.graph[v1]]
+    let checked = []
+    while(toCheck.length > 0){
+      Object.keys(toCheck[0]['paths']).forEach((neighbor) => {
+        if(!checked.includes(neighbor)){
+          toCheck.push(this.graph[neighbor])
+        }
+      })
+      checked.push(toCheck.shift())
+    }
+    checked.forEach((vertex) => action(vertex))
+  }
+
+  findShortestPath(v1,v2){
```

Comment:
I did not check to see if this works but here is how I would refactor your function while trying to keep all the same functionality

```js
findShortestPath(fromVertName, toVertName){
  let possibleSolutions = [{path:[fromVertName], weight:0}]
  let validSolutions = []

  while(possibleSolutions.length > 0){
    let solution = possibleSolutions.shift()
    let lastVertName = solution.path[solution.path.length - 1]

    if (lastVertName === toVertName){
      validSolutions.push(solution)
      continue;
    }

    let lastVert = this.graph[lastVertName]
    let neighbors = Object.keys(lastVert.paths)

    neighbors.forEach(neighbor => {
      if (solution.path.includes(neighbor)) return
      possibleSolutions.push({
        path:   solution.path.concat(neighbor),
        weight: solution.weight + (lastVert.paths[neighbor] || 0)
      })
    })
  }

  return validSolutions.sort((a, b) => a.weight - b.weight)[0].path
}
```


I tried to dry up a TON of duplication. Look for duplication and store those results in variables with clear names. It will make your code a ton easier to understand.